src/pages/Sheet.tsx

```diff
@@ -1,20 +1,28 @@
-import React from "react";
+import React, { useEffect } from "react";
 import { ILocation } from "../@types/location";
 import { Citizen } from "../components/Citizen";
 import TransactionsList from "../components/TransactionsList";
+import { setCitizenId } from "../helpers/coredata";
 
 type Props = {
   location: ILocation;
 };
 
 const Sheet = (props: Props) => {
+  useEffect(() => {
+    if (props.location.state) {
```

Comment:
this is not very clear. if you have `props.location.state.citizenid` valued here, why do you want to set it in sessionStorage as well? can't you just use  `props.location.state.citizenid` and pass it down to Citizen component?
If the Citizen component does not have citizenid I'd redirect the user to the Sheet route so she can start again
(and avoiding to sync / user sessionStorage at all)

Reply:
what about this?

Reply:
Can we optimize this passing [props.location.state] as useEffect second argument?

Reply:
Pushed in this PR :)